hackerpet/__init__.py

```diff
@@ -0,0 +1,494 @@
+            return cls(int(mode_str))
+        else:
+            return HubMode._missing_(mode_str)
+
+
+@unique
+class HubStatus(Enum):
+    """
+    HubStatus describes the current state of the hub.
+
+    PLAYING: food is in the hub, and player can play
+    EMPTY: no food in hub, player cannot play
+    PLATTER_JAMMED: platter is jammed
+    POD_JAMMED: pod is jammed
+    DOME_REMOVED: dome has been removed
+    """
+
+    PLAYING = auto()
+    EMPTY = auto()
+    PLATTER_JAMMED = auto()
+    POD_JAMMED = auto()
+    DOME_REMOVED = auto()
+
+    @classmethod
+    def _missing_(cls: type, value: Any) -> "HubStatus":
+        raise IllegalValue(cls, value)
+
+    @staticmethod
+    def from_status(status: str) -> "HubStatus":
+        """converts message from hub to Status, or UNKNOWN if invalid"""
+
+        if "Your hub is working" in status:
+            return HubStatus.PLAYING
+        elif "Out of food" in status:
+            return HubStatus.EMPTY
+        elif "Platter is jammed" in status:
+            return HubStatus.PLATTER_JAMMED
+        elif "Singulator is jammed" in status:
+            return HubStatus.POD_JAMMED
+        elif "Dome is removed" in status:
+            return HubStatus.DOME_REMOVED
+        else:
+            return HubStatus._missing_(status)
+
+
+@unique
+class Game(IntEnum):
+    """
+    Game describes the specific game levels available to hackerpet.
+
+    See the __str__ def for a brief description of each game level
+    """
+
+    GAME0 = 0
+    GAME1 = 1
+    GAME2 = 2
+    GAME3 = 3
+    GAME4 = 4
+    GAME5 = 5
+    GAME6 = 6
+    GAME7 = 7
+    GAME8 = 8
+    GAME9 = 9
+    GAME10 = 10
+    GAME11 = 11
+
+    def __str__(self) -> str:
+        match self:
+            case Game.GAME0:
+                return "Eating the food"
+            case Game.GAME1:
+                return "Exploring the Touchpads"
+            case Game.GAME2:
+                return "Engaging Consistently"
+            case Game.GAME3:
+                return "Avoiding Unlit Touchpads"
+            case Game.GAME4:
+                return "Learning the Lights"
+            case Game.GAME5:
+                return "Mastering the Lights"
+            case Game.GAME6:
+                return "Responding Quickly"
+            case Game.GAME7:
+                return "Learning Brightness"
+            case Game.GAME8:
+                return "Learning Double Sequences"
+            case Game.GAME9:
+                return "Learning Longer Sequences"
+            case Game.GAME10:
+                return "Matching Two Colors"
+            case Game.GAME11:
+                return "Matching More Colors"
+
+    @classmethod
+    def _missing_(cls: type, value: Any) -> "Game":
+        raise IllegalValue(Game, value)
+
+    @classmethod
+    def from_int_str(cls: type, game_str: str) -> "Game":
+        """convert str of int value to Game, or `GAME_UNKNOWN` if invalid"""
+        if game_str.isdigit():
```

Comment:
Would you consider raise or return an error as early as possible and then let normal flow and a single return value follow.

Reply:
sure, gave that a try. im always torn between this style and if/else style (tho secretly, i always prefer `Option.getOrElse` style when possible)